src/recorder.cc

```diff
@@ -61,8 +69,13 @@ void Recorder::set_dump_count(unsigned int count) {
 Datum* Recorder::NewDatum(std::string title) {
   Datum* d = data_[index_];
   d->title_ = title;
-  d->vals_.resize(1);
-  d->shapes_.resize(1);
+  if (inject_sim_id_) {
```

Comment:
I disagree, but I've expressed my opinion. @rwcarlsen is welcome to comment. Further, we value vertical space and dislike premature optimization. If you'd really like to optimize, another state variable could be added (`resize_`), and it could be set in constructors.